tls/s2n_connection.c

```diff
@@ -1517,3 +1518,29 @@ S2N_RESULT s2n_connection_dynamic_free_in_buffer(struct s2n_connection *conn)
 
     return S2N_RESULT_OK;
 }
+
+/* Marks s2n_ktls_mode enabled for the connection. */
+S2N_RESULT s2n_connection_set_ktls_enabled(struct s2n_connection *conn, s2n_ktls_mode ktls_mode)
+{
+    RESULT_ENSURE_REF(conn);
+
+    /* kTLS I/O functionality is managed by s2n-tls. kTLS cannot be enabled
+     * if the application sets custom I/O. */
+    if ((ktls_mode == S2N_KTLS_MODE_SEND || ktls_mode == S2N_KTLS_MODE_DUPLEX) && !conn->managed_send_io) {
+        return S2N_RESULT_ERROR;
```

Comment:
I think `s2n_errno` won't get set if you just return `S2N_RESULT_ERROR`. Maybe `RESULT_BAIL` or another safety macro could be better here?